cmd/sink/filters/filters.go

```diff
@@ -60,7 +62,11 @@ func (f *Filters) changeFilters(namespaces map[string][]kubearchiveapi.KubeArchi
 	archiveMap := make(map[NamespaceGroupVersionKind]cel.Program)
 	deleteMap := make(map[NamespaceGroupVersionKind]cel.Program)
 	archiveOnDeleteMap := make(map[NamespaceGroupVersionKind]cel.Program)
+	namespacesMap := make(map[string]struct{})
 	for namespace, kacResources := range namespaces {
+		if namespace != constants.SinkFilterGlobalNamespace {
+			namespacesMap[namespace] = struct{}{}
+		}
```

Comment:
I'm not sure this works as namespaces with no resources do not get an entry in the `SinkFilter` object. For Konflux this will result in nothing getting archived as everything is controller in the `ClusterKubeArchiveConfig`.

We may have to change [this line](https://github.com/kubearchive/kubearchive/blob/main/cmd/operator/internal/controller/common.go#L313) as well.

If we do make this change, then you can probably check `sinkFilter.spec.Namespaces` directly, and there will be no need for the namespace set that is created in this PR. We also have to make there sure there are no other side-effects from having this empty entry.  For example, we may end up with extra (empty) entries in the maps in the sink.

Reply:
I think the check can probably use `slices.Contains` rather than a for loop.
```
if slices.Contains(sf.spec.Namespaces, namespace) {
   ...
}
```

Reply:
> but we end up using more memory than we need.

How are we using more memory?  We're using the same amount of memory that's currently in use. In that case it means your solution is adding more memory than we need, unless I am missing something.

> slice.contains runtime is linear, where f.namespaces[namespace] runtime is constant 

Then just use `sinkFilter.spec.Namespaces[namespace]`?  It's a map.

I'm sorry I suggested `slices.Contains`.